From: SeongJae Park <sj@kernel.org>
Date: Tue, 28 Jan 2025 12:56:10 -0800
Subject: [PATCH] mm/damon/core: do intervals tuning before reset_aggregated()

set_attrs(), which can be called from kdamond_tune_intervals(), cannot
be called after reset_aggregated().  Do kdamond_tune_intervals() before
reset_aggregated().

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 44579013b88d..b2f3c93f0bbe 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2376,8 +2376,6 @@ static int kdamond_fn(void *data)
 		if (ctx->passed_sample_intervals >= next_aggregation_sis) {
 			ctx->next_aggregation_sis = next_aggregation_sis +
 				ctx->attrs.aggr_interval / sample_interval;
-
-			kdamond_reset_aggregated(ctx);
 			if (ctx->passed_sample_intervals >=
 					ctx->next_intervals_tune_sis) {
 				ctx->next_intervals_tune_sis +=
@@ -2386,6 +2384,8 @@ static int kdamond_fn(void *data)
 					ctx->attrs.tune_interval_aggrs;
 				kdamond_tune_intervals(ctx);
 			}
+
+			kdamond_reset_aggregated(ctx);
 			kdamond_split_regions(ctx);
 			if (ctx->ops.reset_aggregated)
 				ctx->ops.reset_aggregated(ctx);
-- 
2.39.5

